src/CQRS/LeanCode.CQRS.Security/UserIdExtractor.cs

```diff
@@ -0,0 +1,30 @@
+using System.Security.Claims;
+
+namespace LeanCode.CQRS.Security;
+
+public interface IUserIdExtractor<TUserId> : IUserIdExtractor
+    where TUserId : notnull, IEquatable<TUserId>
+{
+    new TUserId Extract(ClaimsPrincipal user);
+}
+
+public interface IUserIdExtractor
+{
+    string Extract(ClaimsPrincipal user);
+}
+
+public class UserIdExtractor : IUserIdExtractor<Guid>
```

Comment:
The name is too generic compared to the implementation. This handles `Guid`s only, and we use also `string`s, and typed IDs.

We should have a ready-made implementation for all three types of user IDs that we see in our projects (so - `Guid`, `string` and source-generated typed ID, which will be tricky). Plus we should have an extension method that registers it in the DI container.